From 124573809b091bc6193e3f988d26619f1f137f6d Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Sat, 5 Dec 2020 20:45:25 +0530
Subject: [PATCH 26/40] Revert "ANDROID: Kbuild, LLVMLinux: allow overriding
 clang target triple"

This reverts commit f930fc1bdcfc86d78adff9cfeb4b94d81390e1db.
---
 Makefile                 | 10 +++-------
 scripts/clang-android.sh |  4 ----
 2 files changed, 3 insertions(+), 11 deletions(-)
 delete mode 100755 scripts/clang-android.sh

diff --git a/Makefile b/Makefile
index 1aff2d4b7ff9..c91ba213c355 100644
--- a/Makefile
+++ b/Makefile
@@ -549,13 +549,9 @@ endif
 
 ifneq ($(shell $(CC) --version 2>&1 | head -n 1 | grep clang),)
 ifneq ($(CROSS_COMPILE),)
-CLANG_TRIPLE	?= $(CROSS_COMPILE)
-CLANG_TARGET	:= --target=$(notdir $(CLANG_TRIPLE:%-=%))
-ifeq ($(shell $(srctree)/scripts/clang-android.sh $(CC) $(CLANG_TARGET)), y)
-$(error "Clang with Android --target detected. Did you specify CLANG_TRIPLE?")
-endif
-GCC_TOOLCHAIN_DIR := $(dir $(shell which $(LD)))
-CLANG_PREFIX	:= --prefix=$(GCC_TOOLCHAIN_DIR)
+CLANG_FLAGS	+= --target=$(notdir $(CROSS_COMPILE:%-=%))
+GCC_TOOLCHAIN_DIR := $(dir $(shell which $(CROSS_COMPILE)elfedit))
+CLANG_FLAGS	+= --prefix=$(GCC_TOOLCHAIN_DIR)$(notdir $(CROSS_COMPILE))
 GCC_TOOLCHAIN	:= $(realpath $(GCC_TOOLCHAIN_DIR)/..)
 endif
 ifneq ($(GCC_TOOLCHAIN),)
diff --git a/scripts/clang-android.sh b/scripts/clang-android.sh
deleted file mode 100755
index 9186c4f48576..000000000000
--- a/scripts/clang-android.sh
+++ /dev/null
@@ -1,4 +0,0 @@
-#!/bin/sh
-# SPDX-License-Identifier: GPL-2.0
-
-$* -dM -E - </dev/null 2>&1 | grep -q __ANDROID__ && echo "y"
-- 
2.17.1

